crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -387,6 +383,82 @@ impl QueryBuilder {
         Ok(query)
     }
 
+    fn multi_get_events<'a>(
+        cursor: Option<(i64, i64)>,
+        descending_order: bool,
+        limit: i64,
+        filter: Option<EventFilter>,
+    ) -> Result<events::BoxedQuery<'a, Pg>, Error> {
+        let mut query = events::dsl::events.into_boxed();
+        if let Some(cursor_val) = cursor {
+            if descending_order {
+                query = query.filter(events::dsl::tx_sequence_number.lt(cursor_val.0));
+                query = query.filter(events::dsl::event_sequence_number.lt(cursor_val.1));
+            } else {
+                query = query.filter(events::dsl::tx_sequence_number.gt(cursor_val.0));
+                query = query.filter(events::dsl::event_sequence_number.gt(cursor_val.1));
+            }
+        }
+        if descending_order {
+            query = query.order((
+                events::dsl::tx_sequence_number.desc(),
+                events::dsl::event_sequence_number.desc(),
+            ));
+        } else {
+            query = query.order((
+                events::dsl::tx_sequence_number.asc(),
+                events::dsl::event_sequence_number.asc(),
+            ));
+        }
+        query = query.limit(limit + 1);
+
+        if let Some(filter) = filter {
+            if let Some(sender) = filter.sender {
+                let subquery = tx_senders::dsl::tx_senders
+                    .filter(tx_senders::dsl::sender.eq(sender.into_vec()))
+                    .select(tx_senders::dsl::tx_sequence_number);
```

Comment:
Are we using this instead of the `events` table's `senders` array because this is a more efficient form of query? (If so, can we add a comment here explaining that?